packages/nimble-components/src/table/specs/table-columns-hld.md

```diff
@@ -325,69 +325,78 @@ const numberWithUnitCellView = NumberWithUnitCellView.compose({
-TODO: Being actively discussed in: https://github.com/ni/nimble/pull/1153#discussion_r1160996644
-
-Finally, here is a column element that allows a user to register a callback for a click event on a button inside the cell template:
+Because the cell view is not a descendant of the column element, we must add special support for the column to handle events that originate in the cells. A column will define a `delegatedEvents` property which is an array of event names that should be delegated. As this is a static property, it will be part of the `ColumnInternalsOptions` passed to the base column constructor.
 
 ```TS
-type TableColumnButtonCellData = TableStringField<'id'>;
-
-public class TableColumnButton extends TableColumn<TableColumnButtonCellData> {
+export interface ColumnInternalsOptions {
     ...
+    readonly delegatedEvents: readonly string[];
+    ...
+}
 
-    public cellRecordFieldNames = ['id'] as const;
-
-    @attr
-    public idKey: string;
-
-    public getDataRecordFieldNames(): string[] {
-        return [valueKey];
-    }
-
-    public cellViewTag = 'table-cell-view-button';
-
-    public validateCellData(cellData: TCellRecord): void {
-        if (typeof(cellData['id']) !== 'string') {
-            throw new Error('Type for cellData is incorrect!');
-        }
+export class ColumnInternals<TColumnConfig> {
+    ...
+    public readonly delegatedEvents: readonly string[];
+    ...
+    public constructor(options: ColumnInternalsOptions) {
+        this.delegatedEvents = options.delegatedEvents;
+        ...
     }
 }
 
-class ButtonCellView extends TableCellView<TableColumnButtonCellData> {
-    public onButtonClick(): void {
-        this.$emit('button-click', { data: this.cellRecord.id })
+AnchorTableColumn extends TableColumn {
+    constructor() {
+        super({
+            delegatedEvents: ['click'],
+            ...
+        });
     }
+    ...
 }
-const buttonCellView = ButtonCellView.compose({
-    baseName: 'table-cell-view-button',
-    template: html<ButtonCellView>`
-        <nimble-button @click="${(x) => x.onButtonClick()}">
-            <span>Press Me</span>
-        </nimble-button>`,
-    styles: /* styling */
-});
 ```
 
-Angular template:
+Upon connecting a cell view to the DOM, we will attach an event listener for each event type in `columnDelegatedEvents`. The handler will wrap the original event in a new `CustomEvent` and dispatch that to the column.
 
-```HTML
-<nimble-table (button-click)="doSomething($event)">
-    <nimble-table-column-button></nimble-table-column-button>
-</nimble-table>
+```TS
+this.addEventListener(delegatedEventName, event => {
+    this.column.dispatchEvent(new CustomEvent('delegated-event', {
+        details: {
+            originalEvent: event
+        }
+    }));
+});
 ```
 
-Angular code:
+The `CustomEvent`'s details will be of type `DelegatedEventEventDetails`:
 
 ```TS
-private doSomething(CustomEvent e): void {
-    ...
+export interface DelegatedEventEventDetails {
```

Comment:
Since the event is that "an event was delegated," I think it does make sense to have "event" in the name. But yeah, it makes this interface name awkward.

Reply:
I thought about it for a bit but didn't come up with anything better.